packages/compile-common/src/profiler/updated.ts

```diff
@@ -108,50 +101,43 @@ function findUpdatedFiles(
   });
 
   return sourceFiles
-    .map((sourceFile, index) => {
-      const sourceFileStat = sourceFileStats[index];
+    .filter((sourceFile, index) => {
+      const stat = sourceFileStats[index];
 
       // Ignore updating artifacts if source file has been removed.
-      if (sourceFileStat == null) return;
+      if (stat == null) return;
 
       const artifactsUpdatedTime =
-        sourceFilesArtifactsUpdatedTimes[sourceFile] || 0;
+        sourceArtifactsUpdatedTimes[sourceFile] || 0;
       const sourceFileUpdatedTime = (
-        sourceFileStat.mtime || sourceFileStat.ctime
-      ).getTime();
+        stat.mtimeMs || stat.ctimeMs
+      ); 
 
       if (sourceFileUpdatedTime > artifactsUpdatedTime) return sourceFile;
```

Comment:
Since you're changing this from a map to a filter, I'd like to suggest changing this line to match, even if the current line technically works.  As in, I think this should read `return sourceFileUpdatedTime > artifactsUpdatedTime;`.

Reply:
By the way, did we need to change it to `mtimeMs`? I think this change was prompted by an assumption that turned out not to be true, right?

Reply:
Oh gosh! For sure @haltman-at !

Reply:
@eggplantzzz I changed the comparison to use `fs.Stat` `xtimeMs` variant because they're already numbers (number of milliseconds since the POSIX Epoch) instead of calling `mtime.getTime()` (mtime is a Date object)